cherrypy/test/test_states.py

```diff
@@ -161,9 +161,14 @@ def test_1_Restart(self):
         self.assertEqual(db_connection.running, False)
         self.assertEqual(len(db_connection.threads), 0)
 
+    # @pytest.mark.skipif(os.name == 'nt', 'see pytest-dev/pytest-forked#44')
+    @pytest.mark.forked
+    @pytest.mark.xfail(bool(os.getenv('CI')), reason='KeyboardInterrupt #1873')
     def test_2_KeyboardInterrupt(self):
         # Raise a keyboard interrupt in the HTTP server's main thread.
         # We must start the server in this, the main thread
+        if os.name == 'nt':
+            return  # Workaround for the pytest.mark.skipif() above.
```

Comment:
Such cases should use a `pytest.xfail()` call but I think it's a wrong way to go around the conditional run here.

It's better to declare a marker conditionally as shown in the first two posts @ https://github.com/pytest-dev/pytest-forked/issues/44.

Reply:
Yes but then the xfail never happens.  The examples do not deal with an xfail.

Reply:
What do you mean? Why not?

Reply:
We need three marks, not two.

Reply:
You can apply the decorators right after the test method declaration. That would be roughly:
```python
if not IS_WINDOWS:
    test_2_KeyboardInterrupt = pytest.mark.forked(test_2_KeyboardInterrupt)
```